### PR TITLE
Updating DefaultRiskAnalysis to check transaction size

### PR DESCRIPTION
**Pull request by stonecoldpat**
I mentioned this a week or two ago - got around to downloading github etc. 

DefaultRiskAnalysis should be able to warn the user if a transaction they seen on the network or have received via an external method of communication - if the transaction IsStandard() or not - one of the checks includes the size of the transaction. 

More can be done to this class, but I have did this small change for now to understand how GitHub etc works. Once I get time i'll try and update it to take into account all the IsStandard() rules. 

If I have done something wrong let me know - this is all magic to me. 

Paddy

### CODE REVIEW

File: core/src/test/java/org/bitcoinj/wallet/DefaultRiskAnalysisTest.java [hunk 1]
```diff
@@ -142,6 +142,22 @@ public void nonStandardDust() {
         edgeCaseTx.addOutput(DefaultRiskAnalysis.MIN_ANALYSIS_NONDUST_OUTPUT, key1); // Dust threshold
         assertEquals(RiskAnalysis.Result.OK, DefaultRiskAnalysis.FACTORY.create(wallet, edgeCaseTx, NO_DEPS).analyze());
     }
+    
+    @Test
+    public void nonStandardExceedSize() {
+        Transaction nonStandardTx = new Transaction(params);
+        nonStandardTx.addInput(params.getGenesisBlock().getTransactions().get(0).getOutput(0));
+        
+        /*
```
Comment (schildbach) [inline core/src/test/java/org/bitcoinj/wallet/DefaultRiskAnalysisTest.java]: Can you make this a one-line comment?

**Comment by schildbach** [comment]
Thanks, generally I like the idea. Can you clean up your commits a bit?
1) Your second commit only fixes the first. Squash both into one, an interactive rebase helps you with that.
2) Identation: You're mixing tabs and spaces at some points. We're using spaces only, 4 at a time.
3) One space between language keywords like if/for/while and the opening brace.

Somewhere I read that max transaction size is half of the max block size. We're at 100 kB. Is that still correct?

**Comment by stonecoldpat** [comment]
Hopefully i've made the changes right. The max transaction size is still 100k, from the reference code: 

/** The maximum size for transactions we're willing to relay/mine */
static const unsigned int MAX_STANDARD_TX_SIZE = 100000;

I'll try and figure out how to do the interactive rebase shortly. Just had an attempt which failed miserably :(. I'll update the request once it has been rebased.

**Comment by mikehearn** [comment]
The way to do it is something like:

git fetch origin
git rebase -i origin/master

Then edit the on-screen instructions so every line except the top starts with "fixup" instead of "pick". Then save and quit, and see what happens ....

**Comment by stonecoldpat** [comment]
I'll close this pull request and make another one. I have tried to do the fixup/pick rebasing, but it does not seem to have beared any fruit and it is now telling me to resolve conflicts when I do it. Sorry for the hassle.

**Comment by mikehearn** [comment]
OK. Git can be pretty fiddly at first, for sure. Don't worry if you run into problems again, I can do the rebase for you. But it's a good skill to learn :-)

**Pull request closed without merging**